src/Service/TenantFactory.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace App\Service;
+
+use App\Entity\Tenant;
+use App\Repository\TenantRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Query\QueryException;
+
+class TenantFactory
+{
+    public function __construct(private EntityManagerInterface $entityManager, private TenantRepository $tenantRepository)
+    {
+    }
+
+    /**
+     * Get Tenants from array og tenent keys. Create new Tenants
+     * for unknown keys.
+     *
+     * @param array $tenantKeys
+     *
+     * @return array
+     *
+     * @throws QueryException
+     */
+    public function getTenants(array $tenantKeys): array
```

Comment:
getTenants is a bit weird naming since it has side-effects. Maybe setupTenants...